### PR TITLE
Tyler Waite's Module_6 initial commit of updates

### PR DESCRIPTION
**Pull request by tylerw8**
First commit of updates to assignment files.

### CODE REVIEW

File: js/tests/enhancedDate.spec.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      it("should be a string and should return Tuesday", function(){
+          expect(enhancedDate.getDayName(testDate)).to.be.a("string").and.to.equal("Tuesday");
+      });
+  });
+
+
+//Get Month Name Test  
+  describe("Get Month Name", function(){
+    
+      it("should be a string and should return June", function(){
```
Comment (jamdon) [inline js/tests/enhancedDate.spec.js:23]: Same deal with this one - break it into two separate tests.

File: js/tests/enhancedDate.spec.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          expect(enhancedDate.getMonthName(testDate)).to.be.a("string").and.to.equal("June");
+      });
+  });
+
+
+//Is Today Test  
+  describe("Is Today", function(){
+    
+      it("testDate should not be a string and should return false ", function(){
```
Comment (jamdon) [inline js/tests/enhancedDate.spec.js:32]: Break this one into two separate tests as well. Additionally, for functions that return a boolean value, I would test both a date that results in false and a date that results in true.

File: js/tests/enhancedDate.spec.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+
+//Is Future Test 
+  describe("Is Future", function(){ 
+
+      it("testDate should return true", function(){
+          expect(enhancedDate.isFuture(testDate)).to.be.true;
+      });
+    
+      it("should not be a string ", function(){
```
Comment (jamdon) [inline js/tests/enhancedDate.spec.js:50]: A better second test here would be to use another date to test the function returns false under appropriate conditions.

File: js/tests/enhancedDate.spec.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  });
+
+
+//Get Date Test  
+  describe("Get Date", function(){
+      it("should return true that it is an instance of Date", function(){
+          expect(enhancedDate.getDate(testDate)).is.an.instanceof(Date);
+      });
+    
+      it("should return true that it is not a Boolean", function(){
```
Comment (jamdon) [inline js/tests/enhancedDate.spec.js:63]: If you've already validated the return value of a function, testing the return value again is pointless. Instead, I'd recommend testing the two scenarios for what kind of value this function can return - an object or a millisecond value.

File: js/tests/enhancedDate.spec.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+//  var enhancedDate;
+//  
+//  beforeEach(function () {
+//      enhancedDate = new enhancedDate();
+//  });
+
+//Get Day Name Test  
+  describe("Get Day Name", function(){
+    
+      it("should be a string and should return Tuesday", function(){
```
Comment (jamdon) [inline js/tests/enhancedDate.spec.js:14]: Break this into two, separate tests.

**Comment by jamdon** [comment]
Comments inline. Let me know if you want it graded as is.

File: js/tests/enhancedDate.spec.js (hunk 3 above)
Comment (tylerw8) [inline js/tests/enhancedDate.spec.js:50]: I tried doing that, but I kept getting errors when I tried to create a test
date for today and a test date for a past date, and errors if I used an
empty function to test for today.  Not sure why, so I went with these for
now.

**Comment by tylerw8** [comment]
Let me know what the as is grade is and then I'll tell you. =) I probably
have time to do some updates tonight, but I will probably focus on Module 5
first.

**Comment by jamdon** [comment]
I can always just grade it first and you can submit updates for re-grading if you want to do it that way. Some students just want feedback before I grade, so I offer the option. It's up to you.

**Comment by tylerw8** [comment]
Just grade them first and if I have time I'll update the code.

File: js/tests/enhancedDate.spec.js (hunk 3 above)
Comment (jamdon) [inline js/tests/enhancedDate.spec.js:50]: To do it, you would need to call enhanced.setDate() with another value inside the test spec:

`it("should return true", function () { enhancedDate.setDate(/* some value that makes it true */); expect(enhancedDate.isFuture()).to.be.true; });`

then, for the next case:

`it("should return false", function () { enhancedDate.setDate(/* some value that makes it false */); expect(enhancedDate.isFuture()).to.be.false; });`

**Comment by jamdon** [comment]
:+1:

**Pull request closed without merging**